app/services/application/ApplicationService.scala

```diff
@@ -298,4 +314,49 @@ trait ApplicationService extends EventSink with CurrentSchemeStatusHelper {
         None
     }.toList
   }
+
+  def currentSchemeStatusWithFailureDetails(applicationId: String): Future[Seq[SchemeEvaluationResultWithFailureDetails]] = {
+
+    def getOnlineTestEvaluation(repository: OnlineTestEvaluationRepository) =
+      repository.getPassMarkEvaluation(applicationId).map(_.result).recoverWith { case _ => Future.successful(Nil) }
+
+    for {
+      phase1Evaluation <- getOnlineTestEvaluation(phase1EvaluationRepository)
+      phase2Evaluation <- getOnlineTestEvaluation(phase2EvaluationRepository)
+      phase3Evaluation <- getOnlineTestEvaluation(phase3EvaluationRepository)
+      fsacEvaluation <- fsacRepo.getFsacEvaluatedSchemes(applicationId).map(_.getOrElse(Nil).toList)
+      fsbEvaluation <- fsbRepo.findByApplicationId(applicationId).map(_.map(_.evaluation.result).getOrElse(Nil))
+      evaluations = ListMap(
+        "online tests" -> phase1Evaluation,
```

Comment:
I think we should move these into an enum before it starts getting out of hand. 🤔 `ApplicationPhases` or whatever

Reply:
Feels a bit like a premature optimisation at this point, since this is the only place it's required - how would you recommend structuring this enum?

Reply:
Ok @henricook , I'll let this one slide 👀

Reply:
You're a nice guy, guy